### PR TITLE
Update refund & readme

### PR DESCRIPTION
**Pull request by ankitdas13**
API Link : https://razorpay.com/docs/api/refunds/
reference : https://github.com/razorpay/razorpay-php/blob/master/documents/refund.md

### CODE REVIEW

File: resources/refund_test.go [hunk 1]
```diff
@@ -41,3 +41,19 @@ func TestRefundCreate(t *testing.T) {
 	assert.Equal(t, err, nil)
 	utils.TestResponse(jsonByteArray, []byte(fixture), t)
 }
+
+func TestRefundEdit(t *testing.T) {
+	url := constants.REFUND_URL + "/" + TestPaymentID
```
Comment (Madeyedexter) [inline resources/refund_test.go:46]:
```suggestion
	url := constants.REFUND_URL + "/" + TestPaymentID
```

Define a separate constant for `TestRefundID`

File: resources/refund.go [hunk 2]
```diff
@@ -28,3 +28,10 @@ func (refund *Refund) Fetch(refundID string, queryParams map[string]interface{},
 func (refund *Refund) Create(data map[string]interface{}, extraHeaders map[string]string) (map[string]interface{}, error) {
 	return refund.Request.Post(constants.REFUND_URL, data, extraHeaders)
 }
+
+// Edit updates the Refund having the given refundID.
+func (refund *Refund) Edit(refundID string, queryParams map[string]interface{}, extraHeaders map[string]string) (map[string]interface{}, error) {
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply resources/refund.go]: changed edit to `Update`

File: documents/refund.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
+## Refunds
+
+### Create a normal refund
+
+```go
+
+data := map[string]interface{}{
+    "speed": "normal",
+	"notes": map[string]interface{}{
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply documents/refund.md]: added in edit and fetch

File: resources/refund_test.go (hunk 1 above)
Comment (ankitdas13) [reply resources/refund_test.go]: done

File: resources/refund.go [hunk 4]
```diff
@@ -28,3 +28,10 @@ func (refund *Refund) Fetch(refundID string, queryParams map[string]interface{},
...
+
+	url := fmt.Sprintf("%s/%s", constants.REFUND_URL, refundID)
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply resources/refund.go]: added in edit and fetch

File: resources/refund.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Madeyedexter) [reply resources/refund.go]: @ankitdas13 Modify the comment above the method name as well

**Review by Madeyedexter — APPROVED** [review]

**Pull request merged**